src/Composer/DefinitionMutator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nusje2000\ComposerMonolith\Composer;
+
+use LogicException;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class DefinitionMutator implements DefinitionMutatorInterface
+{
+    /**
+     * @var array<mixed, mixed>
+     */
+    protected $originalDefinition;
+
+    /**
+     * @var array<mixed, mixed>
+     */
+    protected $mutatedDefinition;
+
+    /**
+     * @var SplFileInfo
+     */
+    protected $fileInfo;
+
+    public function __construct(SplFileInfo $fileInfo)
+    {
+        $decoded = json_decode($fileInfo->getContents(), true);
+
+        $this->originalDefinition = $decoded;
+        $this->mutatedDefinition = $decoded;
+        $this->fileInfo = $fileInfo;
+    }
+
+    public function setDependency(string $dependency, string $versionConstraint): void
+    {
+        $this->mutatedDefinition['require'][$dependency] = $versionConstraint;
```

Comment:
It's safer to use an object for this instead of a nested array.

Reply:
That is correct but it has some disadvantages that I then have to find a work around for:
1. Comparing the mutated version to the original
2. Getting require-dev since this contains a `-`

I personally think that the use of variable variables is difficult to follow so in array context this would make a bit more sense.